src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -436,14 +436,19 @@ static inline void pio_gpio_init(PIO pio, uint pin) {
     gpio_set_function(pin, pio == pio0 ? GPIO_FUNC_PIO0 : GPIO_FUNC_PIO1);
 }
 
-/*! \brief Return the DREQ to use for pacing transfers to a particular state machine
+/*! \brief Return the DREQ to use for pacing transfers to/from a particular state machine FIFO
  *  \ingroup hardware_pio
  *
  * \param pio The PIO instance; either \ref pio0 or \ref pio1
  * \param sm State machine index (0..3)
  * \param is_tx true for sending data to the state machine, false for received data from the state machine
  */
 static inline uint pio_get_dreq(PIO pio, uint sm, bool is_tx) {
+    static_assert(DREQ_PIO0_TX1 - DREQ_PIO0_TX0 == 1, "");
```

Comment:
Only a minor consistency nit-pick, but in all the other functions added in this PR you do
```
static_assert(CONSTANT_A == CONSTANT_B + INTEGER
```
but here you've done
```
static_assert(CONSTANT_A - CONSTANT_B == INTEGER
```
?